src/fidocadj/FidoFrame.java

```diff
@@ -731,107 +714,69 @@ public FidoFrame createNewInstance()
-
-        circuitPanel.getMapCoordinates().setMagnitudes(options.zoomValue,
-                                                       options.zoomValue);
-        circuitPanel.getMapCoordinates().setXGridStep(options.gridSize);
-        circuitPanel.getMapCoordinates().setYGridStep(options.gridSize);
-
-        ae.setPcbThickness(options.pcblinewidth_i);
-        ae.setPcbPadSizeX(options.pcbpadwidth_i);
-        ae.setPcbPadSizeY(options.pcbpadheight_i);
-        ae.setPcbPadDrill(options.pcbpadintw_i);
-
-        circuitPanel.getDrawingModel().setTextFont(options.macroFont_s,
-            options.macroSize_i,
-            circuitPanel.getUndoActions());
-
-        circuitPanel.setStrictCompatibility(options.extStrict);
-        toolBar.setStrictCompatibility(options.extStrict);
-        cpa.setShiftCopyPaste(options.shiftCP);
-
-        libDirectory=options.libDirectory;
-
-        Globals.lineWidth = options.stroke_size_straight_i;
-        Globals.lineWidthCircles = options.stroke_size_straight_i;
-        Globals.diameterConnection = options.connectionSize_i;
-
-        // We know that this code will be useful only when FidoCadJ will run as
-        // a standalone application. If it is used as an applet, this would
-        // cause the application crash when the applet security model is active.
-        // In this way, we can still use FidoCadJ as an applet, even with the
-        // very restrictive security model applied by default to applets.
-
-        if (runsAsApplication) {
-            preferences.put("DIR_LIBS", libDirectory);
-            preferences.put("MACRO_FONT",
-                    circuitPanel.getDrawingModel().getTextFont());
-
-            preferences.put("MACRO_SIZE",
-                    ""+circuitPanel.getDrawingModel().getTextFontSize());
-
-            preferences.put("STROKE_SIZE_STRAIGHT", ""+Globals.lineWidth);
-            preferences.put("STROKE_SIZE_OVAL", ""+Globals.lineWidthCircles);
-            preferences.put("CONNECTION_SIZE", ""+Globals.diameterConnection);
-            preferences.put("SMALL_ICON_TOOLBAR",
-                    smallIconsToolbar?"true":"false");
-
-            preferences.put("TEXT_TOOLBAR", textToolbar?"true":"false");
-            preferences.put("GRID_SIZE",
-                    ""+circuitPanel.getMapCoordinates().getXGridStep());
-
-            // Save default PCB characteristics
-            preferences.put("pcbPadSizeX", ""+ae.pcbPadSizeX);
-            preferences.put("pcbPadSizeY", ""+ae.pcbPadSizeY);
-            preferences.put("pcbPadStyle", ""+ae.pcbPadStyle);
-            preferences.put("pcbPadDrill", ""+ae.pcbPadDrill);
-            preferences.put("pcbThickness", ""+ae.pcbThickness);
-            preferences.put("SHIFT_CP", cpa.getShiftCopyPaste()?"true":"false");
-
-        }
-        if(!libDirectory.equals(oldDirectory)) {
+        AddElements ae = eea.getAddElements();
+
+        // Creare e visualizzare la finestra di dialogo per le impostazioni
```

Comment:
Do not leave comments in Italian

Reply:
I'll fix them right away, just some temporary comments that slipped through...